src/viking.js

```diff
@@ -1,14 +1,110 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor (health, strength) {
+    this.health = health
+    this.strength = strength
+  }
+
+  attack() {
+    return this.strength
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor (name, health, strength) {
+    super (health, strength) 
+    this.name = name
+  }
+
+  attack() {
+    return this.strength
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`
+    } else {
+      return `${this.name} has died in act of combat`
+    }
+  }
+
+  battleCry() {
+    return "Odin Owns You All!"
+  }
+
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage) {
+    this.health -= damage
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`
+    } else {
+      return `A Saxon has died in combat`
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor () {
+    this.vikingArmy = []
+    this.saxonArmy = []
+  }
+
+  addViking(viking) {
+    this.vikingArmy.push(viking)
+  }
+
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon)
+  }
+
+  vikingAttack() {
+    let randomSaxon = this.saxonArmy[Math.floor(Math.random() *this.saxonArmy.length)]
+    let randomViking = this.vikingArmy[Math.floor(Math.random() *this.vikingArmy.length)]
+
+    let combatResult = randomSaxon.receiveDamage(randomViking.strength)
+
+    if (randomSaxon.health <= 0) {
+      this.saxonArmy.splice(this.saxonArmy.indexOf(randomSaxon), 1)
+    }
+
+    return combatResult
+  }
+
+  saxonAttack() {
+    let randomViking = this.vikingArmy[Math.floor(Math.random() *this.vikingArmy.length)]
+    let randomSaxon = this.saxonArmy[Math.floor(Math.random() *this.saxonArmy.length)]
+
+    let combatResult = randomViking.receiveDamage(randomSaxon.strength)
+
+    if (randomViking.health <= 0) {
+      this.vikingArmy.splice(this.vikingArmy.indexOf(randomViking), 1)
+    }
+
+    return combatResult
+  }
```

Comment:
Está bien lo que has implementado, pero con un filter también se podría realizar. Te lo pongo a continuación y así tienes otra manera de verlo!!
` saxonAttack() {
const viking = randomElement(this.vikingArmy);
const saxon = randomElement(this.saxonArmy);

const attackResult = viking.receiveDamage(saxon.strength);
this.vikingArmy = this.vikingArmy.filter(viking => viking.health > 0);
return attackResult;
}`